pypesto/objective/aggregate.py

```diff
@@ -0,0 +1,175 @@
+import numpy as np
+
+from .objective import Objective
+
+
+class AggregateObjective(Objective):
+    """
+    This class allows to create an aggregateObjective from a list of
+    Objective instances.
+    """
+
+    def __init__(self, objectives, x_names=None, options=None):
+        """
+        Constructor.
+
+        Parameters
+        ----------
+
+        objectives: list
+            List of pypesto.objetive instances
+
+        """
+        # input typechecks
+        if not isinstance(objectives, list):
+            raise TypeError(f'Objectives must be a list, '
+                            f'was {type(objectives)}.')
+
+        if not all(
+                isinstance(objective, Objective)
+                for objective in objectives
+        ):
+            raise TypeError('Objectives must only contain elements of type'
+                            'pypesto.Objective')
+
+        self.objectives = objectives
```

Comment:
picky, but one should also check that ``len(objectives) > 0`` for various reasons.